platform/ios/src/MGLMapAccessibilityElement.mm

```diff
@@ -49,24 +50,30 @@ - (instancetype)initWithAccessibilityContainer:(id)container feature:(id<MGLFeat
         NSString *languageCode = [MGLVectorSource preferredMapboxStreetsLanguage];
         NSString *nameAttribute = [NSString stringWithFormat:@"name_%@", languageCode];
         NSString *name = [feature attributeForKey:nameAttribute];
-        
+
         // If a feature hasn’t been translated into the preferred language, it
         // may be in the local language, which may be written in another script.
-        // Romanize it.
-        NSLocale *locale = [NSLocale localeWithLocaleIdentifier:languageCode];
-        NSOrthography *orthography;
-#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunguarded-availability-new"
-        if ([NSOrthography respondsToSelector:@selector(defaultOrthographyForLanguage:)]) {
-            orthography = [NSOrthography defaultOrthographyForLanguage:locale.localeIdentifier];
-        }
-#pragma clang diagnostic pop
-#endif
-        if ([orthography.dominantScript isEqualToString:@"Latn"]) {
-            name = [name stringByApplyingTransform:NSStringTransformToLatin reverse:NO];
+        // Attempt to transform to the script of the preferred language, keeping
+        // the original string if no transform exists or if transformation fails.
+        if (@available(iOS 9.0, *)) {
+            NSStringTransform transform;
+            NSString *dominantScript = [NSOrthography mgl_dominantScriptForMapboxStreetsLanguage:languageCode];
+
+            if ([dominantScript isEqualToString:@"Latn"]) {
+                transform = NSStringTransformToLatin;
+            } else if ([dominantScript isEqualToString:@"Hans"]) {
+                // Transform not available.
+            } else if ([dominantScript isEqualToString:@"Cyrl"]) {
+                transform = NSStringTransformLatinToCyrillic;
+            } else if ([dominantScript isEqualToString:@"Arab"]) {
+                transform = NSStringTransformLatinToArabic;
```

Comment:
FYI, a feature in Russia may not have been localized into an Arabic-script language, in which case two steps would be required: `Any-Latn`, then `Latn-Arab`. NSString accepts transform identifiers other than the ones provided through the NSStringTransform constants. [This document](http://userguide.icu-project.org/transforms/general) explains the transform identifiers and how to use them.

Reply:
Thanks! That’s very cool, I didn’t realize there were more transformations hiding under the hood.

I adopted `Any-Latin; Any-Arabic` (and `Cyrillic`) in https://github.com/mapbox/mapbox-gl-native/pull/10881/commits/1da24da97059f5b0d081bb26efa6b648abfe141e#diff-3d5f0d34b6286db4d72ebfe0dbd161aaR53.

Reply:
Incidentally, [here’s the official list of transforms](http://icu-project.org/charts/comparison/transforms.html). Unfortunately, ones such as `Any-Arabic` don’t work for `Cyrillic → Arabic`, presumably because there’s no direct `Cyrillic-Arabic`.